File: core/src/com/broll/gainea/server/core/utils/StreamUtils.java
package com.broll.gainea.server.core.utils;

import java.util.function.Consumer;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public final class StreamUtils {

    private StreamUtils() {

    }

    public static <T> void safeForEach(Stream<T> stream, Consumer<? super T> consumer) {
        copy(stream).forEach(consumer);
    }

    public static <T> Stream<T> copy(Stream<T> stream) {
        return stream.collect(Collectors.toList()).stream();
    }
}

File: server/src/main/com/broll/gainea/server/core/utils/StreamUtils.kt
package com.broll.gainea.server.core.utils

import java.util.function.Consumer
import java.util.stream.Collectors
import java.util.stream.Stream

object StreamUtils {
    fun <T> safeForEach(stream: Stream<T>, consumer: Consumer<in T>?) {
        copy(stream).forEach(consumer)
    }

    fun <T> copy(stream: Stream<T>): Stream<T> {
        return stream.collect(Collectors.toList()).stream()
    }
}
